src/features/users.ts

```diff
@@ -0,0 +1,22 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export const loadUsers = createAsyncThunk('users/fetch', async () => {
+  const result = await getUsers().then(usersFromServer => usersFromServer);
+
+  return result;
+});
+
+export const usersSlice = createSlice({
+  name: 'users',
+  initialState: [] as User[],
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(loadUsers.fulfilled, (_, actions) => {
+      return actions.payload;
```

Comment:
It's a good practice to not mutate the state directly even in redux-toolkit. Consider using immer's `produce` function or spread operator to return a new state. For example: `return [...actions.payload];`